gooddata-java/src/test/java/com/gooddata/sdk/service/hierarchicalconfig/HierarchicalConfigServiceIT.java

```diff
@@ -0,0 +1,135 @@
+    private static final String SETTING_ITEM_NAME = "myCoolFeature";
+    private static final String PROJECT_SETTING_URI_STRING = PROJECT_SETTING_TEMPLATE
+            .expand(PROJECT_ID, SETTING_ITEM_NAME).toString();
+
+    @BeforeMethod
+    public void setUp() throws Exception {
+        service = gd.getHierarchicalConfigService();
+        project = readObjectFromResource("/project/project.json", Project.class);
+    }
+
+    @Test
+    public void projectSettingsShouldReturnProjectSettings() {
+        onRequest()
+                .havingMethodEqualTo("GET")
+                .havingPathEqualTo(PROJECT_SETTINGS_URI_STRING)
+                .respond()
+                .withBody(readStringFromResource("/hierarchicalconfig/settings.json"))
+                .withStatus(200);
+
+        final Settings settings = service.projectSettings(project);
+
+        assertThat(settings, containsInAnyOrder(
+                new SettingItem("myCoolFeature", "true", "project"),
+                new SettingItem("mySuperCoolFeature", "true", "project"),
+                new SettingItem("mySuperSecretFeature", "false", "project"),
+                new SettingItem("myNotBooleanSetting", "Disabled", "catalog")));
+    }
+
+    @Test
+    public void createProjectSettingShouldCreateAndReturnSameSetting() {
+        final String settingItemUriString = PROJECT_SETTING_TEMPLATE.expand(project.getId(), SETTING_ITEM_NAME).toString();
+
+        onRequest()
+                .havingMethodEqualTo("PUT")
+                .havingPathEqualTo(PROJECT_SETTING_URI_STRING)
+                .respond()
+                .withStatus(201);
+        onRequest()
+                .havingMethodEqualTo("GET")
+                .havingPathEqualTo(settingItemUriString)
+                .respond()
+                .withBody(readStringFromResource("/hierarchicalconfig/projectSetting.json"))
+                .withStatus(200);
+
+        final SettingItem settingItem = service.upsertProjectSetting(project,
+                new SettingItem(SETTING_ITEM_NAME, "true", "project"));
+
+        assertThat(settingItem, is(notNullValue()));
+        assertThat(settingItem.getName(), is(SETTING_ITEM_NAME));
+        assertThat(settingItem.isEnabled(), is(true));
+        assertThat(settingItem.getSource().getSource(), is("project"));
+    }
+
+    @Test
+    public void getSettingShouldReturnSetting() {
+        onRequest()
+                .havingMethodEqualTo("GET")
+                .havingPathEqualTo(PROJECT_SETTING_URI_STRING)
+                .respond()
+                .withBody(readStringFromResource("/hierarchicalconfig/projectSetting.json"))
+                .withStatus(200);
+
+        final SettingItem item = service.getProjectSetting(project, SETTING_ITEM_NAME);
+
+        assertThat(item, is(new SettingItem(SETTING_ITEM_NAME, "true", "project")));
+    }
+
+    @Test
+    public void updateSettingShouldUpdateAndReturnUpdatedSetting() {
```

Comment:
I would join it with method checking creation if it's now being done by one method.

Reply:
Joined